cmd/rtc-utils/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/TheCacophonyProject/rtc-utils/rtc"
+	"github.com/alexflint/go-arg"
+)
+
+func main() {
+	err := runMain()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+var version = "<not set>"
+
+type Args struct {
+	Read         bool `arg:"--read" help:"read RTC to system time"`
+	WriteIfSync  bool `arg:"--write-if-sync" help:"write system time to RTC if NTP is synchronized"`
+	Write        bool `arg:"--write" help:"write system time to RTC"`
+	CheckBattery bool `arg:"--check-battery" help:"check if the RTC battery is low"`
+	Attempts     int  `args:"--attempts" help:"number of times to try reading/writing registers to the RTC"`
+}
+
+func (Args) Version() string {
+	return version
+}
+
+func procArgs() Args {
+	args := Args{
+		Attempts: 1,
+	}
+	p := arg.MustParse(&args)
+	if !args.CheckBattery && !args.Read && !args.Write && !args.WriteIfSync {
+		p.Fail("no options given")
+	}
```

Comment:
Given that exactly one is required, you could use a single positional argument to decide on the command to run. In this scheme, only `--attempts` would be an option.

Example command lines 

```
rtc read
rtc write
rtc write-if-sync
rtc check-battery
rtc --attempts=3 check-battery
```

go-arg supports positional arguments:https://github.com/alexflint/go-arg#positional-arguments. Just read into a `Mode` argument and fail if it's not one of the supported commands. This would simplify the code too.